docs/services/ml/upload-training-script.md

```diff
@@ -0,0 +1,216 @@
+                    coords.append(
+                        [
+                            annotation["y_min_normalized"],
+                            annotation["x_min_normalized"],
+                            annotation["y_max_normalized"],
+                            annotation["x_max_normalized"],
+                        ]
+                    )
+            bbox_labels.append(labels)
+            bbox_coords.append(coords)
+    return image_filenames, bbox_labels, bbox_coords
+```
+
+{{% /expand%}}
+{{%expand "Step 3: Add logic to produce model artifact" %}}
+
+After reading in the dataset and preparing the data for training, you can add logic to build and compile the model and train the model on the data.
+The code in this part of your script depends on the type of ML training you are doing.
+
+{{% /expand%}}
+{{%expand "Step 4: Save the model artifact the script produces" %}}
+
+You also need to include in your <file>training.py</file> logic to save the model artifact your training job produces to the `model_output_directory`.
+For example:
+
+```python
+# This allows us to save the model artifact to Viam,
+# which will be viewable as a registry item for the ML model name
+# and version specified
+def save_tflite_classification(
+    model: Model,
+    model_dir: str,
+    model_name: str,
+    target_shape: ty.Tuple[int, int, int],
+) -> None:
+    # Convert the model to tflite, with batch size 1
+    # so the graph does not have dynamic-sized tensors.
+    input = tf.keras.Input(target_shape, batch_size=1, dtype=tf.uint8)
+    output = model(input, training=False)
+    wrapped_model = tf.keras.Model(inputs=input, outputs=output)
+    converter = tf.lite.TFLiteConverter.from_keras_model(wrapped_model)
+    converter.target_spec.supported_ops = TFLITE_OPS
+    tflite_model = converter.convert()
+
+    # Save the model to GCS
+    filename = os.path.join(model_dir, f"{model_name}.tflite")
+    with open(filename, "wb") as f:
+        f.write(tflite_model)
+```
+
+{{% /expand%}}
+
+### Package the training script as a <file>tar.gz</file> source distribution
+
+Follow the instructions to [create a <file>tar.gz</file> gzip'd tar file](https://docs.python.org/3.10/distutils/sourcedist) from your project.
+
+{{% alert title="Tip" color="tip" %}}
+You can reference the directory structure of this [example classification training script](https://app.viam.com/packages/e76d1b3b-0468-4efd-bb7f-fb1d2b352fcb/custom-training-classification/ml_training/latest/e76d1b3b-0468-4efd-bb7f-fb1d2b352fcb).
+See <file>model/training.py</file> for an example entrypoint file.
+{{% /alert %}}
+
+## Upload a new training script or new version
+
+You must use the Viam CLI to upload your training script to the Registry.
+You can use the [`viam training-script upload`](/cli/#training-script) command in the form `viam training-script upload --path=<path-to-tarball> --org-id=<your-org-id> --script-name=<name-for-script>` to upload a new script.
```

Comment:
```suggestion
You must use the Viam CLI to upload your training script to the Registry.

To upload a training script us the [`viam training-script upload`](/cli/#training-script) command:

```sh {class="command-line" data-prompt="$" data-output="1-10"}
viam training-script upload --path=<path-to-tarball> --org-id=<your-org-id> --script-name=<name-for-script>
```

```